core/schema/git.go

```diff
@@ -103,88 +109,183 @@ type gitArgs struct {
 	SSHAuthSocket dagql.Optional[core.SocketID] `name:"sshAuthSocket"`
 }
 
-func (s *gitSchema) git(ctx context.Context, parent *core.Query, args gitArgs) (*core.GitRepository, error) {
+func (s *gitSchema) git(ctx context.Context, parent dagql.Instance[*core.Query], args gitArgs) (inst dagql.Instance[*core.GitRepository], _ error) {
```

Comment:
@grouville suggestion for a follow-up - is there a way to break down some of this logic into separate functions, maybe even into `core/git.go` itself?

The comments here *really* help, but if we could use function calls to help show the structure that would be even better :tada:

Reply:
Yeah sure 👍 I saw a lot of monolithic functions, so wasn't sure ; but 👌